include/elf/arc-reloc.def

```diff
@@ -497,3 +497,24 @@ ARC_RELOC_HOWTO(ARC_NPS_CMEM16, 78, \
                 replace_bits16, \
                 dont, \
                 ( ME ( S + A )))
+
+ARC_RELOC_HOWTO(ARC_JLI_SECTOFF, 63, \
+                2, \
+                10, \
+                replace_jli, \
+                bitfield, \
+                ( ( S - JLI ) ))
+
+ARC_RELOC_HOWTO(ARC_JLI_SECTOFF_SIMM12, 79, \
+                2, \
+                12, \
+                replace_add2_jli, \
+                bitfield, \
+                ( ( S - JLI ) ))
+
+ARC_RELOC_HOWTO(ARC_JLI_32, 80, \
+                2, \
+                32, \
+                replace_word32_jli, \
+                bitfield, \
+                ( ( S - JLI ) ))
```

Comment:
In this case, yes (which I have changed to simply S and using replace_word32); however, if the program is being loaded and relocated the loading code may wish to use JLI_BASE + ( S - JLI ) which could differ from the section's address. Here, JLI_BASE would be read from the AUX register. For this reason I wanted a different relocation type.